Tests/Twig/Extension/SonataHelpersExtensionTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Twig\Extension;
+
+use Sonata\AdminBundle\Twig\Extension\SonataHelpersExtension;
+use Symfony\Bridge\Twig\Tests\Extension\Fixtures\StubFilesystemLoader;
+
+/**
+ * Defined here for brevity.
+ */
+class StringLike
+{
+    public function __toString()
+    {
+        return 'string-like';
+    }
+}
+
+class SonataHelpersExtensionTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var SonataHelpersExtension
+     */
+    private $twigExtension;
+
+    /**
+     * @var \Twig_Environment
+     */
+    private $environment;
+
+    public function setUp()
+    {
+        $loader = new StubFilesystemLoader(array(
+            __DIR__.'/../../../Resources/views/CRUD',
+        ));
+        $this->environment = new \Twig_Environment($loader, array(
+            'strict_variables' => true,
+            'cache'            => false,
+            'autoescape'       => false,
+            'optimizations'    => 0,
+        ));
+        $this->twigExtension = new SonataHelpersExtension();
+        $this->environment->addExtension($this->twigExtension);
+        $this->twigExtension->initRuntime($this->environment);
+    }
+
+    /**
+     * @dataProvider testRenderClasslistProvider
+     */
+    public function testRenderClassList($input, $expected)
+    {
+        $output = call_user_func_array(array($this->twigExtension, 'renderClasslist'), $input);
+        $this->assertSame($expected, $output);
+    }
+
+    public function testRenderClasslistProvider()
+    {
+        $stringLike = new StringLike();
+        $iterable = new \ArrayObject(array('foo' => true, 'bar' => true, 'baz' => false));
+
+        return array(
+            // classlist('foo') is a shortcut to classlist({foo: true})
+            array(array('foo', 'bar'), 'foo bar'),
+            array(array(array('foo' => true, 'bar' => true)), 'foo bar'),
+            // ignores falsy values
```

Comment:
Instead of using comments, you can use them directly as the indices of the array : If the test fails for a given index, the string will appear in the error message :)